### PR TITLE
[13.0][FIX] cmis_field: fix backend_name parameter

### PR DESCRIPTION
**Pull request by jdoutreloux**
Fix the _init_ function to allow the use of a second backend.

### CODE REVIEW

File: cmis_field/fields/cmis_folder.py [hunk 1]
```diff
@@ -72,12 +73,18 @@ class CmisFolder(fields.Field):
...
-        super(CmisFolder, self).__init__(
-            backend_name=backend_name, string=string, **kwargs)
+    def __init__(self, string=None, **kwargs):
+        super(CmisFolder, self).__init__(string=string, **kwargs)
 
     def _description_backend(self, env):
         backend = self.get_backend(env, raise_if_not_found=False)
+        if len(backend) > 1:
+            if any(i.function == "load_modules" for i in inspect.stack()):
+                backend = backend[0]
```
Comment (lmignon) [inline cmis_field/fields/cmis_folder.py]: @jdoutreloux  Good idea :tada:. IMO this condition should be put into a dedicated method `if self._is_registry_loading_mode()`. Nevertheless, I'm a little bit afraid by the performance cost of such a method where we always inspect all the call stack. We should also deduce that we are in the module loading process by checking the `_lock` attribute on the registry.

```python
def  _is_registry_loading_mode(self):
    """ Check if we are in the registry loading process.

    Before loading the modules, a reentrant lock is acquired
    by the registry. By checking the count of acquired lock
    we can deduce that the registry is in a incoherent state
    """
    return self.env.registry._lock._RLock__count == 0

def _description_backend(self, env):
    backend = self.get_backend(env, raise_if_not_found=False)
    if lself._is_registry_loading_mode():
        # While the registry is loading, specific attributes are no available
        # on the field (such as `backend_name`). At this stage, the fields
        # are accessed to validate the xml views of the module being 
        # loaded/updated. We can therefore safely takes the first backend
        # into the list.
        backend = backend[:1]
```

File: cmis_field/fields/cmis_folder.py [hunk 2]
```diff
@@ -72,12 +73,39 @@ class CmisFolder(fields.Field):
...
-            backend_name=backend_name, string=string, **kwargs)
+    def __init__(self, string=None, **kwargs):
+        super(CmisFolder, self).__init__(string=string, **kwargs)
+
+    def  _is_registry_loading_mode(self, env):
+        """ Check if we are in the registry loading process.
+
+        Before loading the modules, a reentrant lock is acquired
+        by the registry. By checking the count of acquired lock
+        we can deduce that the registry is in a incoherent state
```
Comment (sbidoul) [inline cmis_field/fields/cmis_folder.py]: This comment seem out of date ?

File: cmis_field/fields/cmis_folder.py [hunk 3]
```diff
@@ -72,12 +73,39 @@ class CmisFolder(fields.Field):
...
+        if len(backend) > 1:
+            if self._is_registry_loading_mode(env):
+                # While the registry is loading, specific attributes are no available
+                # on the field (such as `backend_name`). At this stage, the fields
+                # are accessed to validate the xml views of the module being 
+                # loaded/updated. We can therefore safely takes the first backend
+                # into the list.
+                backend = backend[:1]
+            else:
+                import pdb; pdb.set_trace()
```
Comment (sbidoul) [inline cmis_field/fields/cmis_folder.py]: pdb ;)

**Review by sbidoul — COMMENTED** [review]
Tricky stuff :)

File: cmis_field/fields/cmis_folder.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 import threading
+import inspect
```
Comment (lmignon) [inline cmis_field/fields/cmis_folder.py]: @jdoutreloux This import is useless :smirk:

**Comment by lmignon** [comment]
@jdoutreloux Can you fix the conflicts so we can merge...

**Comment by sbejaoui** [comment]
rebased

**Review by sbejaoui — APPROVED** [review]

**Pull request merged**